cloudprem/variables.tf

```diff
@@ -233,4 +254,30 @@ variable "environment" {
     condition     = length(var.environment) <= 5
     error_message = "The length of the Environment must be less than 6 characters."
   }
+}
+variable "stack_type" {
+  description = "Specify whether this is a production stack or a development stack. Production stacks have data protection settings enabled but will leave resources behind after a terraform delete operation."
+  type = string
+  default = "production"
+
+  validation {
+    condition = var.stack_type == "dev" || var.stack_type == "production"
```

Comment:
Just a quick thought... maybe we can use a [terraform function](https://www.terraform.io/docs/language/functions/contains.html) to check against a list of good values? That could help make this rule a little more declarative for maintenance?

Reply:
I'll definitely look into that. This variable definition was created by our contractors.